diff --git a/src/lib/TransactionBuilder/TransactionBuilder.ts b/src/lib/TransactionBuilder/TransactionBuilder.ts
index a395086..fe19e63 100644
--- a/src/lib/TransactionBuilder/TransactionBuilder.ts
+++ b/src/lib/TransactionBuilder/TransactionBuilder.ts
@@ -2417,8 +2417,17 @@ export class TransactionBuilder {
             if (typeof transaction.visible === 'boolean') {
                 res.transaction.transaction.visible = transaction.visible;
             }
+            const _transaction = res.transaction.transaction;
+            /* @ts-expect-error ignore */
+            if(transaction.payInfo){
+                /* @ts-expect-error ignore */
+                _transaction.payInfo = {
+                    /* @ts-expect-error ignore */
+                    ...transaction.payInfo
+                }
+            }
             return resultManager(
-                res.transaction.transaction as unknown as TransactionWrapper,
+                _transaction as unknown as TransactionWrapper,
                 {
                     ...transaction.raw_data.contract[0].parameter.value,
                     Permission_id: transaction.raw_data.contract[0].Permission_id,
diff --git a/src/lib/TransactionBuilder/helper.ts b/src/lib/TransactionBuilder/helper.ts
index f1b45ca..2a56235 100644
--- a/src/lib/TransactionBuilder/helper.ts
+++ b/src/lib/TransactionBuilder/helper.ts
@@ -40,6 +40,14 @@ export function resultManagerTriggerSmartContract(
     if (!(options._isConstant || options.estimateEnergy)) {
         const authResult = txCheckWithArgs(transaction.transaction, data, options);
         if (authResult) {
+            /* @ts-expect-error ignore */
+            if(data && data.function_selector){
+                /* @ts-expect-error ignore */
+                transaction.transaction.payInfo = {
+                    /* @ts-expect-error ignore */
+                    function_selector: data.function_selector
+                }
+            }
             return transaction;
         }
         throw new Error('Invalid transaction');
